### ISSUE TITLE
Include Purchase Worksheet Feature to DMS

### ISSUE THREAD

**Issue by RoyalDrive** [Royal-Drive/DMS#121]
[Royal Drive_Worksheet_v2.3.pdf](https://github.com/Royal-Drive/DMS/files/8492157/Royal.Drive_Worksheet_v2.3.pdf)

This PDF was created custom for closing our sales. 

The idea would be to consolidate the number of apps/platforms by being able to generate these documents from the DMS. Doing so would require us to add the following inputs seen below to the DMS in the "Customer" section. Adding a few more input boxes and a button to generate the documents would (hopefully) not be that difficult a project. This would be a second-tier task directly under the role-based access. 

![image](https://user-images.githubusercontent.com/79367027/163474463-f1795415-1a0e-4c8c-986c-53898ef6d3bd.png)

**Comment by RoyalDrive** [comment]
![163474463-f1795415-1a0e-4c8c-986c-53898ef6d3bd](https://user-images.githubusercontent.com/79367027/182865985-750cad71-2405-42b4-b821-6bf946a1f48e.png)

**Comment by RoyalDrive** [comment]
The unnecessary features are lined out. Everything else is needed but may not on every contract like a credit card number or trade-in.

**Comment by JacobWeisenburger** [comment]
if we go with the picture as background option, then you might be using more ink when you print it. I can remove most of the white in the background of the pages, but it still might use more ink. It might end up being not a noticeable difference though. Just something to consider

**Comment by RoyalDrive** [comment]
Good point. A trial run would be awesome to make it work/looks okay/doesn't use my whole ink cartridge lol. Are you able to give one page a shot with the Purchase Worksheet? Or do you want me to send over each document separately like how I originally sent it to the guy who designed the PDFs?

**Comment by RoyalDrive** [comment]
Ya know, super random thought, but the guy who makes these PDFs is named Matthew Calendar and he told me over the phone while back that he was wanting to make his PDF creations into a webapp. Are you thinking this could be a good collaborative project if we rope him in? Or would you prefer to keep it simple and just keep the work in-house?

**Comment by RoyalDrive** [comment]
Nope I sent that message before we did the print test. It looks great - let's proceed without Matthew Calendar as planned,

**Comment by JacobWeisenburger** [comment]
> Nope I sent that message before we did the print test. It looks great

sounds good

---
> let's proceed without Matthew Calendar as planned

Unless he knows a way for a PDF to take text inputs programmatically. Because then he could add that functionality to your current PDF, and all we have to do is connect it to the DMS.

**Comment by RoyalDrive** [comment]
I can email him and ask?

**Comment by JacobWeisenburger** [comment]
couldn't hurt to ask, unless he is going to charge you for that answer. lol

Also I found a code library that claims to be able to fill out pdf forms from JavaScript, so I'll look into it some more, but this could be exactly what we are needing.
https://pdf-lib.js.org/

**Comment by JacobWeisenburger** [comment]
I haven't tried it yet, but I think I should be able to use https://pdf-lib.js.org/ to fill out the form that you already have.

---
Or if you wanted to make a new version of that PDF, then here are some helpful links. I could work on this if you want, or if you wanted to save money, you could use these videos to make your own thing and save it in the royal drive google drive, that way you can make changes in the future without needing to hire that out.
[How To Create a Fillable PDF Form For FREE! - YouTube](https://www.youtube.com/watch?v=9m8O0-n2X-s)
[How to Create Fillable PDF Form for FREE - YouTube](https://www.youtube.com/watch?v=6cYpJJxvZMc)

---
Thoughts?

**Comment by RoyalDrive** [comment]
I like the look and format of the PDFs we use now. If you can use those - great! If you can't, I can have Christa design a PDF from scratch with Canva, she has a pro account with them.

**Comment by JacobWeisenburger** [comment]
Sounds good. I'll do some testing tomorrow or next week.

**Comment by RoyalDrive** [comment]
Okay, I'll wait to contact Matt Calendar since we've got some options here.
